assume/common/outputs.py

```diff
@@ -569,3 +574,29 @@ def get_sum_reward(self):
         rewards_by_unit = np.array(rewards_by_unit)
 
         return rewards_by_unit
+
+    def write_flows(self, data: any):
```

Comment:
data looks like `dict[tuple[datetime, str], float]`, is this correct?

Reply:
That is correct. This is the output of promo. I decided to leave it like that and do the data format handling in the output role to keep the market role as decluttered as possible.